api-hub/v1beta1/managedclustermodule_webhook.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var modulelog = logf.Log.WithName("managed-cluster-module-resource")
+
+func (mcm *ManagedClusterModule) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	// controller-runtime will set the path to `validate-<group>-<version>-<resource> so we
+	// need to make sure it is set correctly in the +kubebuilder annotation below.
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(mcm).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-hub-kmm-sigs-x-k8s-io-v1beta1-managedclustermodule,mutating=false,failurePolicy=fail,sideEffects=None,groups=hub.kmm.sigs.x-k8s.io,resources=managedclustermodules,verbs=create;update,versions=v1beta1,name=vmanagedclustermodule.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &ManagedClusterModule{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (mcm *ManagedClusterModule) ValidateCreate() error {
+	modulelog.Info("Validating ManagedClusterModule creation", "name", mcm.Name, "namespace", mcm.Namespace)
+
+	module := &kmmv1beta1.Module{
+		Spec: mcm.Spec.ModuleSpec,
+	}
+	return module.ValidateCreate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (mcm *ManagedClusterModule) ValidateUpdate(obj runtime.Object) error {
+	modulelog.Info("Validating ManagedClusterModule update", "name", mcm.Name, "namespace", mcm.Namespace)
+
+	module := &kmmv1beta1.Module{
+		Spec: mcm.Spec.ModuleSpec,
+	}
+	return module.ValidateUpdate(obj)
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (mcm *ManagedClusterModule) ValidateDelete() error {
+
```

Comment:
Do you we maybe need a log info here as well? Like what we have [here](https://github.com/kubernetes-sigs/kernel-module-management/pull/368/files#diff-fe0b4a96e1db25686e8c4c6b7b46b7c12afb5bbf530ea4f7a63d50aaad4db06eR44) and [here](https://github.com/kubernetes-sigs/kernel-module-management/pull/368/files#diff-fe0b4a96e1db25686e8c4c6b7b46b7c12afb5bbf530ea4f7a63d50aaad4db06eR54) ?

Reply:
The reason we don't have a log is because the validation is "empty" we are not checking anything, which means it is basically like there was no webhook at all when removing. (we are not having a log on deleting `Module`s either in the `Module`'s webhook.

Do you still think we should have a log here?

Reply:
Added the log anyway cause it is calling an internal function and it is not clear that it is not doing anything.